pkg/sql/distsqlplan/aggregator_funcs_test.go

```diff
@@ -33,15 +34,25 @@ import (
 )
 
 var (
-	// compareCtx is a decimal context with reduced precision for comparing decimal
-	// results.  For some decimal aggregate operations there will often be
+	// compareCtx is a decimal context with reduced precision for comparing
+	// decimal results.
+	// For some decimal aggregate operations there will often be
 	// unavoidable off-by-last-digit errors, which is semantically okay.
-	compareCtx = parser.DecimalCtx.WithPrecision(parser.DecimalCtx.Precision - 1)
-	// floatPrecFmt is the format string with a precision of 10 (after
+	compareCtx = &apd.Context{
+		Precision: parser.DecimalCtx.Precision - 1,
+		// We truncate down instead of rounding since the last digit
+		// may cause the new last digit to also be off by one.
+		// For example, if we use the default RoundHalfUp and the last
+		// digits of the decimals we want to compare are 4 and 5,
+		// then the former will be rounded down and the latter will
```

Comment:
Won't this still fail for 0.19 vs 0.20? (both could be valid if the real result is close to 0.195). Seems like the right thing to do is not to truncate but be less strict when comparing.

Reply:
Oh yes, you're correct. I thought it would be ok because of how we handle 

https://github.com/cockroachdb/cockroach/blob/aad470f0cedacd5bff4b7a09a479a2b9c4ab8466/pkg/sql/logictest/testdata/logic_test/aggregate#L809

But that only works because we know the expected output and so can be more confident it won't have the rollover you describe. For reference, here's how apd tests do their 1ulp comparison:

https://github.com/cockroachdb/apd/blob/b1ce49cb2a474f4416531e7395373eaafaa4fbe2/gda_test.go#L672

Reply:
Good point, I'll push out another PR again to remedy this.